models/classes/resources/Contract/InstanceContentCopierInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ *
+ * @author Andrei Shapiro <andrei.shapiro@taotesting.com>
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\resources\Contract;
+
+use core_kernel_classes_Resource;
+
+interface InstanceContentCopierInterface
+{
+    public function copy(
+        core_kernel_classes_Resource $instance,
+        core_kernel_classes_Resource $destinationInstance
+    ): void;
```

Comment:
Returning `void` here seems inconsistent with other interfaces (for example, `ClassPropertyCopierInterface`). Could we return a reference to the instance here too? (and maybe add a PHPDoc comment clarifying that the returned instance is the source).

nitpick: Also, using `$from` and `$to` as parameter names is a bit more clear (they would be the same names used by the the standard [copy()](https://www.php.net/copy) function)

```suggestion
    public function copy(
        core_kernel_classes_Resource $from,
        core_kernel_classes_Resource $to
    ): core_kernel_classes_Resource;
```

Reply:
IMO we should change the other interfaces to return `void` instead. Since `copy` is a **command** and it should not return anything. 

@hectoras the `core_kernel_classes_Resource $destinationInstance` is already the resource you want to return, so returning it is also redundant.

Reply:
I agree, my concern is more about consistency than having the destination returned (however, the return value would be a sort-of syntactic sugar for things like this: https://github.com/oat-sa/extension-tao-item/pull/577/files#r863586257).

In that regard, we'd also need to change [ClassCopier.php](https://github.com/oat-sa/tao-core/pull/3503/files#diff-116193cda220e3b93637a5ded7e849b7922572bc8836dfd01b5e57f978e54271) to not return the created class (btw I think the returned class is not used right now).

Reply:
Then, we may just change the parameter names to match the names from the standard [copy()](https://www.php.net/copy) function (suggestion).

```suggestion
    public function copy(
        core_kernel_classes_Resource $from,
        core_kernel_classes_Resource $to
    ): void;
```

Reply:
I think it's not a good idea to use `from`-`to` for all of them.
This will be ok for cases like:
* `copy class metadata to another class metadata`
* `copy instance metadata to another instance metadata`
* etc.

But in cases like `copy instance to class` or `copy class to class` sounds for me like `something`-`destination` or `something`-`to`.
Because in this case, we are copying `element` to the specific `destination`, not `from element`.

Reply:
Also, I think, that at least for `ClassCopier` and `InstanceCopier` it's better to return new instance.

Imagine, that I copied a class/instance to another class, which contains a billion of classes/instances.
If we return to the user just a destination class - he will be confused, because he needs to search for this class/instance in the list.

As for me, after copying, it's better to provide a link to the copied class/instance, so the user can open it immediately (or we can open this class/instance automatically).
For this, we need to return a copied class/instance.